src/EStimLibrary/Core/ResourceManager.cs

```diff
@@ -40,9 +42,9 @@ public bool TryGetNextAvailableId(out int globalId)
         while (!this.IdPool.TryGetNextFreeId(out globalId))
         {
             // If max capacity already used, return failure.
-            if (!Utils.IsWithinUpperBound(this.IdPool.NumUsedIds,
-                this.MaxNumResources))
+            if (this.IdPool.NumUsedIds >= this.MaxNumResources)
```

Comment:
was the previous implementation failing? If so, should fix that rather than change a line here

Reply:
I understand it now! The implementation of IsWithinUpperBound had "(inclusive)", but according to chatGPT, doing so results in Operator Precedence Issue, which would prioritize like so: "The ternary operator ? : has higher precedence than the logical OR ||.". So, I added parentheses in the return statement of IsWithinUpperBound, and added "inclusive: false" in TryGetNextAvailableId. It passed the test cases